syntax = "proto3";

package embedding;

// Service definition for embedding
service EmbeddingService {
  // Stream the JSON data to another service
  rpc StreamEmbedding (EmbeddingRequest) returns (stream EmbeddingResponse);
}

// Request message containing the text file stream
message EmbeddingRequest {
  bytes file_stream = 1; // Byte stream of the file
}

// Response message containing only the JSON stream
message EmbeddingResponse {
  string json_stream = 1; // JSON data
}
